### PR TITLE
DEV-4744: RevEngine not sending country code to Bad Actor API (Cf-Ipcountry header)

### PR DESCRIPTION
**Pull request by GuiMend**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)
n/a
#### What's this PR do?
Sends user's country_code to bad actor
#### Why are we doing this? How does it help us?
Help identify in bad actor if card issuer and user country are the same
#### Are there detailed, specific, step-by-step testing instructions in the associated ticket?
yes
#### Did this PR make changes to the user interface that may require changes to the user-facing docs?
no
#### Have automated unit tests been added? If not, why?
n/a
#### How should this change be communicated to end users?
n/a
#### Are there any smells or added technical debt to note?
no
#### Has this been documented? If so, where?
no
#### What are the relevant tickets? Add a link to any relevant ones.
https://news-revenue-hub.atlassian.net/browse/DEV-4744
#### Do any changes need to be made before deployment to production (adding environment variables, for example)? If so, open a ticket and link it to the ticket for this PR and list it here:
~~Probably need to remove the `logging` commit added for BadActor to assist in QA.~~ no longer needed
#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:
no

### CODE REVIEW

File: apps/contributions/serializers.py [hunk 1]
```diff
@@ -394,7 +394,9 @@ def get_bad_actor_score(self, data):
             "page": data["page"].id,
             "referer": self.context["request"].META.get("HTTP_REFERER"),
             "ip": get_original_ip_from_request(self.context["request"]),
+            "country_code": self.context["request"].headers.get("Cf-Ipcountry", None),
         }
+        logger.info("BadActorSerializer data: %s", data)
```
Comment (GuiMend) [inline apps/contributions/serializers.py:399]: Needed for QA. If removing this log is better, I can remove the commit right after QA passes.

File: apps/contributions/serializers.py [hunk 2]
```diff
@@ -394,7 +394,9 @@ def get_bad_actor_score(self, data):
             "page": data["page"].id,
             "referer": self.context["request"].META.get("HTTP_REFERER"),
             "ip": get_original_ip_from_request(self.context["request"]),
+            "country_code": self.context["request"].headers.get("Cf-Ipcountry", None),
```
Comment (benjaminEwhite) [inline apps/contributions/serializers.py:397]: Believe it will default to `None` by default.

**Review by benjaminEwhite — CHANGES_REQUESTED** [review]
One small comment inline. Aside from that, I think it would make sense to have a unit test around the business logic here (show that when header has CF-Ipcountry, the data passed to `make_bad_actor_request` contains the value, and when header does not a have an entry, not included in data).

File: apps/contributions/serializers.py (hunk 2 above)
Comment (benjaminEwhite) [reply apps/contributions/serializers.py:397]: Also, the AC say that if no code in request header, then should not be included in request to bad actor.  So need some conditionality around that.

Comment (GuiMend) [reply apps/contributions/serializers.py:397]: Got it, my bad. I thought that sending `"country_code": None` was enough to pass AC2.

**Review by benjaminEwhite — APPROVED** [review]

File: apps/contributions/serializers.py (hunk 1 above)
Comment (x110dc) [reply apps/contributions/serializers.py:403]: I think it's fine to leave it as-is.

**Review by x110dc — APPROVED** [review]

**Review by benjaminEwhite — APPROVED** [review]

**Pull request merged**